File: php/tests/Conformance/AmendFixture/AmendWidget.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Tests\Conformance\AmendFixture;

use SimpleOrm\Metadata\Attributes\Column;
use SimpleOrm\Metadata\Attributes\Generated;
use SimpleOrm\Metadata\Attributes\Key;
use SimpleOrm\Metadata\Attributes\Table;

/**
 * The `diff --amend` conformance fixture (ADR-0017 add.3,
 * conformance/amend-cases/README.md): table `amend_widgets` with a generated
 * key, `name`, and a nullable `remark` — the model has moved on to `remark`
 * since the draft `V0002` (which added `note`) was generated. `V0001`/`V0002`
 * below, and their table steps, are the "compiled" truth `DiffCommand` reads
 * through `MigrationSet::fromDirectory()` — independent of whatever a given
 * case lays into its own ephemeral migrations directory.
 */
#[Table('amend_widgets')]
final class AmendWidget
{
    #[Key]
    #[Generated]
    #[Column]
    public int $id;

    #[Column]
    public string $name;

    #[Column]
    public ?string $remark = null;
}

File: go/orm/internal/conformance/amendfixture/models/amend_widget.go
// Package models is the entity leaf of the amend-cases fixture — split from
// the amendfixture package the same way orm/sample is split from
// orm/sample/migrations: the step package (Table/AmendWidget) must import the
// entity type, and the root package (amendfixture) must import the step
// package, so the entity type cannot live in the root package too without an
// import cycle (a Go constraint the C#/PHP ports don't hit).
package models

import "github.com/kosmassp/SimpleOrm/go/orm"

// AmendWidget is the conformance/amend-cases fixture entity (ADR-0017 add.3,
// conformance/amend-cases/README.md), matching
// dotnet/tests/SimpleOrm.Tests/AmendFixtures.cs and
// php/tests/Conformance/AmendFixture exactly: the model has moved on to
// Remark since the draft V0002 (which added note) was generated — that
// mismatch is the point of the fixture.
type AmendWidget struct {
	ID     int64   `orm:"column,key,generated"`
	Name   string  `orm:"column"`
	Remark *string `orm:"column"`
}

func (AmendWidget) Entity() orm.EntityDef {
	return orm.EntityDef{Source: orm.Table("amend_widgets")}
}
